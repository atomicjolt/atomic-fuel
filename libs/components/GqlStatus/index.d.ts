declare function GqlStatus(props: any): JSX.Element | null;
declare namespace GqlStatus {
    namespace propTypes {
        const loading: PropTypes.Requireable<boolean>;
        const error: PropTypes.Requireable<PropTypes.InferProps<{
            message: PropTypes.Requireable<string>;
        }>>;
        const loadMessage: PropTypes.Requireable<string>;
    }
}
export default GqlStatus;
import PropTypes from "prop-types";
