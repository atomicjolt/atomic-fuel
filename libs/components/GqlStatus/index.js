"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = GqlStatus;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _atomicjolt_loader = _interopRequireDefault(require("../common/atomicjolt_loader"));

var _inline_error = _interopRequireDefault(require("../common/errors/inline_error"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function GqlStatus(props) {
  if (props.loading) {
    return /*#__PURE__*/_react["default"].createElement(_atomicjolt_loader["default"], {
      message: props.loadMessage
    });
  }

  if (props.error) {
    return /*#__PURE__*/_react["default"].createElement(_inline_error["default"], {
      error: props.error.message
    });
  }

  return null;
}

GqlStatus.propTypes = {
  loading: _propTypes["default"].bool,
  error: _propTypes["default"].shape({
    message: _propTypes["default"].string
  }),
  loadMessage: _propTypes["default"].string
};
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIi4uLy4uLy4uL3NyYy9jb21wb25lbnRzL0dxbFN0YXR1cy9pbmRleC5qc3giXSwibmFtZXMiOlsiR3FsU3RhdHVzIiwicHJvcHMiLCJsb2FkaW5nIiwibG9hZE1lc3NhZ2UiLCJlcnJvciIsIm1lc3NhZ2UiLCJwcm9wVHlwZXMiLCJQcm9wVHlwZXMiLCJib29sIiwic2hhcGUiLCJzdHJpbmciXSwibWFwcGluZ3MiOiI7Ozs7Ozs7QUFBQTs7QUFDQTs7QUFDQTs7QUFDQTs7OztBQUVlLFNBQVNBLFNBQVQsQ0FBbUJDLEtBQW5CLEVBQTBCO0FBQ3ZDLE1BQUlBLEtBQUssQ0FBQ0MsT0FBVixFQUFtQjtBQUNqQix3QkFDRSxnQ0FBQyw2QkFBRDtBQUFrQixNQUFBLE9BQU8sRUFBRUQsS0FBSyxDQUFDRTtBQUFqQyxNQURGO0FBR0Q7O0FBRUQsTUFBSUYsS0FBSyxDQUFDRyxLQUFWLEVBQWlCO0FBQ2Ysd0JBQU8sZ0NBQUMsd0JBQUQ7QUFBYSxNQUFBLEtBQUssRUFBRUgsS0FBSyxDQUFDRyxLQUFOLENBQVlDO0FBQWhDLE1BQVA7QUFDRDs7QUFFRCxTQUFPLElBQVA7QUFDRDs7QUFFREwsU0FBUyxDQUFDTSxTQUFWLEdBQXNCO0FBQ3BCSixFQUFBQSxPQUFPLEVBQUVLLHNCQUFVQyxJQURDO0FBRXBCSixFQUFBQSxLQUFLLEVBQUVHLHNCQUFVRSxLQUFWLENBQWdCO0FBQUVKLElBQUFBLE9BQU8sRUFBRUUsc0JBQVVHO0FBQXJCLEdBQWhCLENBRmE7QUFHcEJQLEVBQUFBLFdBQVcsRUFBRUksc0JBQVVHO0FBSEgsQ0FBdEIiLCJzb3VyY2VzQ29udGVudCI6WyJpbXBvcnQgUmVhY3QgZnJvbSAncmVhY3QnO1xuaW1wb3J0IFByb3BUeXBlcyBmcm9tICdwcm9wLXR5cGVzJztcbmltcG9ydCBBdG9taWNKb2x0TG9hZGVyIGZyb20gJy4uL2NvbW1vbi9hdG9taWNqb2x0X2xvYWRlcic7XG5pbXBvcnQgSW5saW5lRXJyb3IgZnJvbSAnLi4vY29tbW9uL2Vycm9ycy9pbmxpbmVfZXJyb3InO1xuXG5leHBvcnQgZGVmYXVsdCBmdW5jdGlvbiBHcWxTdGF0dXMocHJvcHMpIHtcbiAgaWYgKHByb3BzLmxvYWRpbmcpIHtcbiAgICByZXR1cm4gKFxuICAgICAgPEF0b21pY0pvbHRMb2FkZXIgbWVzc2FnZT17cHJvcHMubG9hZE1lc3NhZ2V9IC8+XG4gICAgKTtcbiAgfVxuXG4gIGlmIChwcm9wcy5lcnJvcikge1xuICAgIHJldHVybiA8SW5saW5lRXJyb3IgZXJyb3I9e3Byb3BzLmVycm9yLm1lc3NhZ2V9IC8+O1xuICB9XG5cbiAgcmV0dXJuIG51bGw7XG59XG5cbkdxbFN0YXR1cy5wcm9wVHlwZXMgPSB7XG4gIGxvYWRpbmc6IFByb3BUeXBlcy5ib29sLFxuICBlcnJvcjogUHJvcFR5cGVzLnNoYXBlKHsgbWVzc2FnZTogUHJvcFR5cGVzLnN0cmluZyB9KSxcbiAgbG9hZE1lc3NhZ2U6IFByb3BUeXBlcy5zdHJpbmcsXG59OyJdfQ==