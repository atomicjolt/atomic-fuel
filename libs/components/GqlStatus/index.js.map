{"version":3,"file":"index.js","sourceRoot":"","sources":["../../../src/components/GqlStatus/index.jsx"],"names":[],"mappings":";;;;;AAAA,gDAA0B;AAC1B,0DAAmC;AACnC,kFAA2D;AAC3D,+EAAwD;AAExD,SAAwB,SAAS,CAAC,KAAK;IACrC,IAAI,KAAK,CAAC,OAAO,EAAE;QACjB,OAAO,CACL,8BAAC,2BAAgB,IAAC,OAAO,EAAE,KAAK,CAAC,WAAW,GAAI,CACjD,CAAC;KACH;IAED,IAAI,KAAK,CAAC,KAAK,EAAE;QACf,OAAO,8BAAC,sBAAW,IAAC,KAAK,EAAE,KAAK,CAAC,KAAK,CAAC,OAAO,GAAI,CAAC;KACpD;IAED,OAAO,IAAI,CAAC;AACd,CAAC;AAZD,4BAYC;AAED,SAAS,CAAC,SAAS,GAAG;IACpB,OAAO,EAAE,oBAAS,CAAC,IAAI;IACvB,KAAK,EAAE,oBAAS,CAAC,KAAK,CAAC,EAAE,OAAO,EAAE,oBAAS,CAAC,MAAM,EAAE,CAAC;IACrD,WAAW,EAAE,oBAAS,CAAC,MAAM;CAC9B,CAAC"}