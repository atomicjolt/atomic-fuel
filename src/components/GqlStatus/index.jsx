import React from 'react';
import PropTypes from 'prop-types';
import AtomicJoltLoader from '../../common/atomicjolt_loader';
import InlineError from '../../common/errors/inline_error';

export default function GqlStatus(props) {
  if (props.loading) {
    return (
      <AtomicJoltLoader message={props.loadMessage} />
    );
  }

  if (props.error) {
    return <InlineError error={props.error.message} />;
  }

  return null;
}

GqlStatus.propTypes = {
  loading: PropTypes.bool,
  error: PropTypes.shape({ message: PropTypes.string }),
  loadMessage: PropTypes.string,
};