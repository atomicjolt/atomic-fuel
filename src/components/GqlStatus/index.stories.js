import React from 'react';

import StoryWrapper from '../StoryWrapper';
import GqlStatus from './index.jsx';

export default {
  title: 'GqlStatus',
  component: GqlStatus
}

const Template = (args) => {
  return (
    <StoryWrapper>
      <GqlStatus {...args} />
    </StoryWrapper>
  );
}

export const Loading = Template.bind({});

Loading.args = {
  loading: true,
  loadingMessage: 'Please wait a moment kind sir'
}
