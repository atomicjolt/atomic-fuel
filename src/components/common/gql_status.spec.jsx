import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import GqlStatus from './gql_status';

const mockStore = configureStore([]);
const store = mockStore({});

describe('GqlStatus', () => {
  it('renders loading', () => {
    const props = {
      loading: true,
      error: {
        message: 'There was an error',
      },
      loadMessage: 'Loading...',
    };
    const { container } = render(
      <Provider store={store}>
        <GqlStatus {...props} />
      </Provider>);
    expect(container).toMatchSnapshot();
  });

  it('renders error', () => {
    const props = {
      loading: false,
      error: {
        message: 'There was an error',
      },
      loadMessage: 'Loading...',
    };
    const { container } = render(
      <Provider store={store}>
        <GqlStatus {...props} />
      </Provider>);
    expect(container).toMatchSnapshot();
  });
});
