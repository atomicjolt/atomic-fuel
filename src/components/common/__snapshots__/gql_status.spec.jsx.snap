// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GqlStatus renders error 1`] = `
<div>
  <div
    class="error-banner"
  >
    <i
      class="material-icons"
    >
      error
    </i>
    <h3>
      Error
    </h3>
    <div
      class="error-banner__content"
    >
      There was an error
    </div>
  </div>
</div>
`;

exports[`GqlStatus renders loading 1`] = `
<div>
  <div
    class="aj-loader"
  >
    <div
      class="atomicjolt-loading-animation"
    >
      <svg
        aria-label="loading"
        role="img"
        viewBox="0 0 91.87 114.09"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          data-name="Layer 2"
        >
          <polygon
            class="cls-1"
            points="40.45 111.32 89.11 99.26 71.35 19.9 21.1 89.71 40.45 111.32"
          />
          <polyline
            class="cls-2"
            points="50.67 2.77 2.77 69.96 25.47 94.65 66.36 84.13 50.67 2.77 71.35 19.9"
          />
        </g>
      </svg>
    </div>
    <p
      class="loader-text"
    >
      Loading...
    </p>
  </div>
</div>
`;
